zilliqa/src/consensus.rs

```diff
@@ -504,16 +502,21 @@ impl Consensus {
                     % self.config.consensus.new_view_broadcast_interval.as_secs())
                     == 0
             {
-                if let Some((_, ExternalMessage::NewView(new_view))) =
-                    self.new_view_message_cache.as_ref()
-                {
-                    if new_view.view == self.get_view()? {
-                        // When re-sending new view messages we broadcast them, rather than only sending them to the
-                        // view leader. This speeds up network recovery when many nodes have different high QCs.
-                        return Ok(Some((None, ExternalMessage::NewView(new_view.clone()))));
+                match self.network_message_cache.take() {
+                    Some((_, ExternalMessage::NewView(new_view))) => {
+                        // If new_view message is not for this view then it must be outdated
+                        if new_view.view == self.get_view()? {
+                            // When re-sending new view messages we broadcast them, rather than only sending them to the
+                            // view leader. This speeds up network recovery when many nodes have different high QCs.
+                            return Ok(Some((None, ExternalMessage::NewView(new_view))));
+                        }
+                    }
+                    Some((peer, ExternalMessage::Vote(vote))) => {
+                        if vote.view == self.get_view()? {
+                            return Ok(Some((peer, ExternalMessage::Vote(vote))));
+                        }
                     }
-                    // If new_view message is not for this view then it must be outdated
-                    self.new_view_message_cache = None;
+                    _ => unimplemented!("network_message_cache"),
```

Comment:
Hm wont this panic  in the case of `self.new_view_message_cache = None`, which is how it is initialised